file/lib/src/sentry_file.dart

```diff
@@ -0,0 +1,387 @@
+    FileMode mode = FileMode.write,
+    bool flush = false,
+  }) {
+    _wrapSync(
+      () => _file.writeAsBytesSync(bytes, mode: mode, flush: flush),
+      'file.write',
+    );
+  }
+
+  @override
+  Future<File> writeAsString(
+    String contents, {
+    FileMode mode = FileMode.write,
+    Encoding encoding = utf8,
+    bool flush = false,
+  }) {
+    return _wrap(
+      () async => _file.writeAsString(
+        contents,
+        mode: mode,
+        encoding: encoding,
+        flush: flush,
+      ),
+      'file.write',
+    );
+  }
+
+  @override
+  void writeAsStringSync(
+    String contents, {
+    FileMode mode = FileMode.write,
+    Encoding encoding = utf8,
+    bool flush = false,
+  }) {
+    _wrapSync(
+      () => _file.writeAsStringSync(
+        contents,
+        mode: mode,
+        encoding: encoding,
+        flush: flush,
+      ),
+      'file.write',
+    );
+  }
+
+  String _getDesc() {
+    return uri.pathSegments.isNotEmpty ? uri.pathSegments.last : path;
+  }
+
+  Future<T> _wrap<T>(Callback<T> callback, String operation) async {
+    final desc = _getDesc();
+
+    final currentSpan = _hub.getSpan();
+    final span = currentSpan?.startChild(operation, description: desc);
+
+    span?.setData('file.async', true);
+    if (_hub.options.sendDefaultPii) {
+      span?.setData('file.path', absolute.path);
+    }
+    T data;
+    try {
+      // workaround for having the length when the file does not exist
+      // or its being deleted.
+      int? length;
+      var hasLength = false;
+      try {
+        length = await _file.length();
+        hasLength = true;
+      } catch (_) {
+        // ignore in case something goes wrong
+      }
+
+      data = await callback();
+
+      if (!hasLength) {
+        try {
+          length = await _file.length();
+        } catch (_) {
+          // ignore in case something goes wrong
+        }
+      }
+
+      if (length != null) {
+        span?.setData('file.size', length);
+      }
+
+      span?.status = SpanStatus.ok();
+    } catch (exception) {
+      span?.throwable = exception;
+      span?.status = SpanStatus.internalError();
+      rethrow;
+    } finally {
+      await span?.finish();
+    }
+    return data;
+  }
+
+  T _wrapSync<T>(Callback<T> callback, String operation) {
+    final desc = _getDesc();
+
+    final currentSpan = _hub.getSpan();
+    final span = currentSpan?.startChild(operation, description: desc);
+    span?.setData('file.async', false);
+
+    if (_hub.options.sendDefaultPii) {
+      span?.setData('file.path', absolute.path);
+    }
+
+    T data;
+    try {
+      // workaround for having the length when the file does not exist
+      // or its being deleted.
+      int? length;
+      var hasLength = false;
+      try {
+        length = _file.lengthSync();
+        hasLength = true;
+      } catch (_) {
+        // ignore in case something goes wrong
+      }
+
+      data = callback() as T;
+
+      if (!hasLength) {
```

Comment:
What if we are overriding the content of the file do we want to know the length before or after?
If I'm looking correctly at the moment we would get the length before.

Reply:
The data can only be read before, because after, the file can be deleted, renamed, moved, so it's a trade-off we made.
Maybe I should point that out in the docs.